src/altinn-app-frontend/src/components/base/CheckboxesContainerComponent.tsx

```diff
@@ -118,37 +124,39 @@ export const CheckboxContainerComponent = ({
       hasSelectedInitial.current === false;
 
     if (shouldSelectOptionAutomatically) {
-      handleDataChange(calculatedOptions[preselectedOptionIndex].value);
+      setValue(calculatedOptions[preselectedOptionIndex].value, true);
       hasSelectedInitial.current = true;
     }
   }, [
     formData?.simpleBinding,
     calculatedOptions,
-    handleDataChange,
+    setValue,
     preselectedOptionIndex,
   ]);
 
   React.useEffect(() => {
     if (optionsHasChanged && formData.simpleBinding) {
       // New options have been loaded, we have to reset form data.
-      // We also skip any required validations
-      handleDataChange(undefined, 'simpleBinding', true);
+      setValue(undefined, true);
```

Comment:
Just adding that this introduces a change where `skipValidation` now is set to false (so custom validation runs in this case, just not required field validation). I'm pretty sure that's what we want though, so I agree this is the right step forward. Same thing in radio groups.